### PR TITLE
Any config path update.

### PR DESCRIPTION
**Pull request by Kwach00**
New Main Features:

- Updated the auto-completion script to complete global paths (starting with `/`) and local paths (starting with `./`).
- Implemented handling of configuration files from any location in the system, passed as an argument to the `mdtool setup motor` command.

### CODE REVIEW

File: template_package/etc/bash_completion.d/mdtool_completion.sh [hunk 1]
```diff
@@ -52,18 +52,22 @@ _mdtool_complete()
 
 	elif [ $COMP_CWORD -eq 4 ]; then
 		if [[ "$prev2" == "motor" ]]; then
+			if [[ "$cur" == ./* || "$cur" == /* ]]; then
```
Comment (jakubmatyszczak) [inline template_package/etc/bash_completion.d/mdtool_completion.sh:55]: if rather easy to do, autocomplete only files with .cfg extension. Now it hints all the filenames.

File: src/ConfigManager.cpp [hunk 2]
```diff
@@ -190,5 +184,84 @@ std::string ConfigManager::validateConfig(std::string configName)
...
+				perror("getcwd() error");
+			}
+#endif
+		}
+	}
+	else
+	{
+		userConfigPath = (userConfigDir + "/" + userConfigPath);
+	}
+	userConfigName = userConfigPath.substr(userConfigPath.find_last_of("/") + 1);
```
Comment (jakubmatyszczak) [inline src/ConfigManager.cpp:266]: when loading config files (from path), only open files with .cfg extension! 

Btw consider check of filesize, if larger than 1M throw error that config seems wrong. Ive accidentally selected sample.txt (1GB random value file) and it added default fields to it and tired to upload it to md80 xd

**Review by jakubmatyszczak — CHANGES_REQUESTED** [review]

File: src/ConfigManager.cpp [hunk 3]
```diff
@@ -190,5 +184,84 @@ std::string ConfigManager::validateConfig(std::string configName)
...
+			if (filename != "." && filename != "..")
+			{
+				defaultFilenames.insert(filename);
+			}
+		}
+		closedir(dir);
+	}
+	else
+	{
+		exit(18886);
```
[the remark this answers is not in the document]
Comment (Kwach00) [reply src/ConfigManager.cpp:214]: That was not my doing... Only change made by me was formatting

File: template_package/etc/bash_completion.d/mdtool_completion.sh (hunk 1 above)
Comment (Kwach00) [reply template_package/etc/bash_completion.d/mdtool_completion.sh:55]: But we want to autocomplete also directories, right? But I'll take a look at it

File: src/ConfigManager.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (jakubmatyszczak) [reply src/ConfigManager.cpp:214]: ok, leave like this

File: template_package/etc/bash_completion.d/mdtool_completion.sh (hunk 1 above)
Comment (jakubmatyszczak) [reply template_package/etc/bash_completion.d/mdtool_completion.sh:55]: only dirs + .cfg, as we only have .cfg motor configs, and .ini is mdtool config

**Pull request merged**